Framework/Kernel/inc/MantidKernel/Unit.h

```diff
@@ -661,6 +661,18 @@ class MANTID_KERNEL_DLL Temperature : public Empty {
 
 //=================================================================================================
 
+MANTID_KERNEL_DLL double timeConversionFloat(std::string input_unit,
+                                             std::string output_unit);
+
+template <typename T>
+void timeConversionVector(std::vector<T> &vec, std::string input_unit,
+                          std::string output_unit) {
+  double factor = timeConversionFloat(input_unit, output_unit);
+  if (factor != 1.0)
+    std::transform(vec.begin(), vec.end(), vec.begin(),
+                   [&factor](T x) -> T { return x * static_cast<T>(factor); });
+}
+
```

Comment:
Maybe unit tests in `UnitTest.h` are missing for this added functionality.

Reply:
thanks, this is indeed missing. I'll mark it as "in progress" fo now.